File: leetcode/389-find-the-difference/main.go
package main

import "fmt"

// XOR solution
func findTheDifference(s string, t string) byte {
	var diff byte
	for i := range s {
		diff ^= s[i]
		diff ^= t[i]
	}
	diff ^= t[len(t)-1]
	return diff
}

func main() {
	fmt.Println(string(findTheDifference("abcd", "abcde")))
}

File: leetcode/389-find-the-difference/main.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        diff = 0
        for i in range(len(s)):
            diff ^= ord(s[i])
            diff ^= ord(t[i])
        diff ^= ord(t[len(t)-1])
        return chr(diff)


solution = Solution()
print(solution.findTheDifference("abcd", "abcde"))
